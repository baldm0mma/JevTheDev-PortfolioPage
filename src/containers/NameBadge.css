.nameBadge-container {
  width: 40%;
  /* background-image: url('../backgrounds/background-light-grey.jpeg');
  background-attachment: fixed;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover; */
  margin-top: 200px;
  margin-bottom: 300px;
  border-radius: 15px;
}