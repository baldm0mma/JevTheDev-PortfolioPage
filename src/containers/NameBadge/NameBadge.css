.nameBadge-container {
  width: 40%;
  margin-top: 200px;
  margin-bottom: 300px;
  border-radius: 15px;
}