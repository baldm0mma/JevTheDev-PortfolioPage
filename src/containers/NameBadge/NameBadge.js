import React from "react";
import "./NameBadge.css";
import Zoom from "react-reveal/Zoom";

export const NameBadge = () => {
  return (
    <Zoom>
      <article className="nameBadge-container">
        <h1 data-testid="page-title">Welcome to _jevTheDev</h1>
        <p data-testid="page-text">
          Hi, I'm Jev. I'm a Dev. This is my portfolio.
        </p>
      </article>
    </Zoom>
  );
};
