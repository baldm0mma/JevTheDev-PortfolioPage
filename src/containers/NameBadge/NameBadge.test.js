import React from "react";
import { shallow } from "enzyme";
import { NameBadge } from "./NameBadge";

describe("NameBadge Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NameBadge />);
  });

  it("should render without crashing", () => {
    expect(wrapper).toBeTruthy();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a welcome message", () => {
    // Approach #1 - assert that shallow wrapper will contain specific React element (exactly as it appears in the code).
    const welcome = <h1 data-testid="page-title">Welcome to _jevTheDev</h1>;
    expect(wrapper).toContainReact(welcome);

    // Approach #2 - assert that a search of shallow wrapper by specific data attribute will render a specific text.
    const welcomeTitle = wrapper.find("[data-testid='page-title']").text();
    expect(welcomeTitle).toBe("Welcome to _jevTheDev");
    const welcomeText = wrapper.find("[data-testid='page-text']").text();
    expect(welcomeText).toBe("Hi, I'm Jev. I'm a Dev. This is my portfolio.");
  });
});
