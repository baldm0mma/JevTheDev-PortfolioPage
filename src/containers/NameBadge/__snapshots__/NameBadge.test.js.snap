// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameBadge Component should match the snapshot 1`] = `
<Zoom>
  <article
    className="nameBadge-container"
  >
    <h1
      data-testid="page-title"
    >
      Welcome to _jevTheDev
    </h1>
    <p
      data-testid="page-text"
    >
      Hi, I'm Jev. I'm a Dev. This is my portfolio.
    </p>
  </article>
</Zoom>
`;
